cmd/loom/gateway_reactors.go

```diff
@@ -32,6 +44,23 @@ func startGatewayReactors(
 	return nil
 }
 
+// Checks if the query server at the given DAppChainReadURI is responding.
+func checkQueryService(name ServiceName, chainID string, DAppChainReadURI string, DAppChainWriteURI string) {
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for ; true; <-ticker.C {
+		rpcClient := client.NewDAppChainRPCClient(chainID, DAppChainWriteURI, DAppChainReadURI)
+		gatewayAddr, err := rpcClient.Resolve(fmt.Sprintf("%s", name))
+		if err != nil {
+			log.Error("Error while resolving service", DAppChainReadURI, err)
```

Comment:
-> `log.Error("Failed to resolve gateway contract", "gateway", name, "err", err)`